code/object/objectsnd.cpp

```diff
@@ -521,17 +521,24 @@ void obj_snd_do_frame()
 		rot_vol_mult = 1.0f;
 		alive_vol_mult = 1.0f;
 		if ( objp->type == OBJ_SHIP ) {
-			if ( !(Ship_info[Ships[objp->instance].ship_info_index].flags & (SIF_BIG_SHIP | SIF_HUGE_SHIP)) ) {
+			ship_info *sip = &Ship_info[Ships[objp->instance].ship_info_index];
+			if ( !(sip->flags & (SIF_BIG_SHIP | SIF_HUGE_SHIP)) ) {
 				if ( objp->phys_info.max_vel.xyz.z <= 0.0f ) {
 					percent_max = 0.0f;
 				}
 				else
 					percent_max = objp->phys_info.fspeed / objp->phys_info.max_vel.xyz.z;
 
-				if ( percent_max >= 0.5f )
-					speed_vol_multiplier = 1.0f;
-				else {
-					speed_vol_multiplier = 0.5f + (percent_max);	// linear interp: 0.5->1.0 when 0.0->0.5
+				if ( sip->min_engine_vol == -1.0f) {
```

Comment:
The comments in this section don't match what the code does (if I read it correctly).

I think for retail the following graph applies:
![msp11151c01ig5i2eab464a00002h1i18g9786f4c74](https://cloud.githubusercontent.com/assets/1013983/7670433/6e28330e-fca4-11e4-9ab2-28c0c1fa1a52.gif)
The volume interpolates between 0.5 and 1 when the ship is between 0 and 50% of its speed. Everything above is capped at 1.


The new option results in the following graph (for `min_engine_vol` = 0.5):
![msp31521d648hf025a1ba60000013a9b94bb92762c0](https://cloud.githubusercontent.com/assets/1013983/7670438/97dafb00-fca4-11e4-852d-54d8d9afd8bb.gif)
I think either linearly interpolating between `min_engine_vol` and `1.0f` based on the speed percentage or following the retail code would be more appropriate here.